test/threeDimLinearProductTest.jl

```diff
@@ -208,13 +251,18 @@ global stdX1 = Statistics.std(pts[:,mask],dims=2)
 @test sum(map(Int, 0.4 .< stdX1[1:3] .< 1.6)) == 3 # had a 2==3 failure here
 @show stdX1[4:6]
 @test sum(map(Int, 0.02 .< stdX1[4:6] .< 0.5)) == 3
+
+
 global muX2 = getPPE(fg, :x2).suggested # Statistics.mean(getVal(fg,:x2),dims=2)
 @show muX2[1:3]-[10.0;0;0]
 @test sum(map(Int, abs.(muX2[1:3]-[10.0;0;0]) .< 1.5)) == 3
 
 # case where points land on trivial rotation error [pi;pi;pi]
 @warn "simplify mean and std testing on Pose3 after #244, see #412"
-pts = getVal(fg,:x2)
+_pts = getVal(fg,:x2)
+_pts = getVal(fg,:x1)
+@cast pts[j,i] := getCoordinates.(Pose3, _pts)[i][j]
```

Comment:
maybe newer code is better, but I previously had issues with calling functions inside the `@cast` line.  To resolve first allocated an array and then just cast that (again just noting for completeness, probably the older code that didnt support)

Reply:
Thanks, will keep it in mind, but I haven't had issues with it yet.